File: .scripts/Models/RadiantOrders/RadiantOrderInfo.ts
export class RadiantOrderInfo {
    public name: string;
    public descriptions: { raw: string; cut: string; sphere: string; mining?: string; };
    public color: [number, number, number];
    public surges: string[];
    public gemstone: string;
    public ideals: RadiantOrderIdeal[];

    constructor(self: Partial<RadiantOrderInfo>) {
        Object.assign(this, self);
    }
}

export class RadiantOrderIdeal {
    public label: string;
    public description: string;
    public quotes: string[];
    public abilities: string[];

    constructor(self: Partial<RadiantOrderIdeal>) {
        Object.assign(this, self);
    }
}

File: tools/Cosmere.Tools/Models/RadiantOrderInfo.cs
namespace Cosmere.Tools.Models;

public class RadiantOrderInfo {
    public string Name { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public string? DefName { get; set; }

    public List<string> Surges { get; set; } = new();

    public List<string> Abilities { get; set; } = new();

    public List<Ideal> Ideals { get; set; } = new();

    public string SprenType { get; set; } = string.Empty;

    public string SprenDescription { get; set; } = string.Empty;

    public List<string> SprenNamePool { get; set; } = new();

    public ColorInfo? Color { get; set; }

    public string Gemstone { get; set; } = string.Empty;
}

public class Ideal {
    public int StormlightMax { get; set; } = 0;

    public string Label { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public List<string> Quotes { get; set; } = new();

    public List<string> Abilities { get; set; } = new();
}
